### ISSUE TITLE
GPU image check if CUDA is available

### ISSUE THREAD

**Issue by AmitMY** [azavea/raster-vision#728]
On the GPU image, when running on CPU there is an error.
Is there a way to add a check for that image "if cuda is available"? It would be more comfortable to work with one image for debugging and production, as I am extending your image

**Comment by lewfish** [comment]
Unfortunately, we haven't found a way to use a single image for both CPU and GPU with TF and those backend libraries. If you find a way, I'd be very interested in hearing about it. Outside of RV, I've been able to use a single image for using PyTorch on CPU and GPU using a check like you're describing.

**Comment by AmitMY** [comment]
Yeah, I'm used to working with PyTorch/DyNet, which both allow choosing the actual device in runtime...

What about https://stackoverflow.com/questions/37660312/how-to-run-tensorflow-on-cpu
